packages/frontend/app/routes/webhooks.tsx

```diff
@@ -0,0 +1,160 @@
+              <h3 className='text-2xl leading-10'>Webhook Events</h3>
+            </div>
+          </PageHeader>
+          <div className='p-1'>
+            <h3 className='text-lg font-bold'>Filters</h3>
+            <div className='flex items-center'>
+              <DropdownFilter
+                label='Type'
+                values={type ? type : ['all']}
+                options={[
+                  {
+                    name: 'All',
+                    value: 'all',
+                    action: () => {
+                      navigate(``)
+                    }
+                  },
+                  ...Object.values(WebhookEventType).map((value) => ({
+                    name:
+                      value.charAt(0).toUpperCase() +
+                      value.slice(1).replace(/[_.]/g, ' '),
+                    value: value,
+                    action: () => {
+                      navigate(`${getTypeFilterParams(type, value)}`)
+                    }
+                  }))
+                ]}
+              />
+            </div>
+          </div>
+          <Table>
+            <Table.Head columns={['ID', 'Type', 'Data']} />
+            <Table.Body>
+              {webhooks.edges.length ? (
+                webhooks.edges.map((webhook) => (
+                  <Table.Row key={webhook.node.id}>
+                    <Table.Cell>{webhook.node.id}</Table.Cell>
+                    <Table.Cell>{webhook.node.type}</Table.Cell>
+                    <Table.Cell>
+                      <Button
+                        aria-label='view webhook data'
+                        state={{ data: webhook.node.data }}
+                        to={'/webhooks/data'}
+                      >
+                        View data
+                      </Button>
+                    </Table.Cell>
+                  </Table.Row>
+                ))
+              ) : (
+                <Table.Row>
+                  <Table.Cell colSpan={4} className='text-center'>
+                    No webhook events found.
+                  </Table.Cell>
+                </Table.Row>
+              )}
+            </Table.Body>
+          </Table>
+          <div className='flex items-center justify-between p-5'>
+            <Button
+              aria-label='go to previous page'
+              disabled={!webhooks.pageInfo.hasPreviousPage}
+              onClick={() => {
+                navigate(
+                  previousPageUrl +
+                    getTypeFilterParams(type, undefined).replace('?', '&')
+                )
+              }}
+            >
+              Previous
+            </Button>
+            <Button
+              aria-label='go to next page'
+              disabled={!webhooks.pageInfo.hasNextPage}
+              onClick={() => {
+                navigate(
+                  nextPageUrl +
+                    getTypeFilterParams(type, undefined).replace('?', '&')
+                )
+              }}
+            >
```

Comment:
We can construct the `nextPageUrl` alongside with the type filters in the `loader` function.

Reply:
I'm not sure I'm following exactly what you want me to change here but I have changed the code a bit as well so that this getTypeFilterParams actually sets the filters now so it would need to be in the component.

Let me know if that resolves your concern or if there is something else I can change.

Reply:
Currently, `nextPageUrl` and `previousPageUrl` are sent from the `loader`. By this meaning, we construct them on the server and then send them to the client.

https://github.com/interledger/rafiki/blob/da0263a44dc774caa0bbceb95ce73331a220a7dc/packages/frontend/app/routes/webhooks.tsx#L32-L38

Instead of performing this operation on the client with `getTypeFilterParams`, we can add all the necessary search params in the loader function as well. As an example, something that have just came into my mind, LOC from 32 to 34 (👆) can be changed to something like this:

```ts
if (webhooks.pageInfo.hasPreviousPage && webhooks.pageInfo.startCursor) {
    const searchParams = new URLSearchParams()
    searchParams.set('before', webhooks.pageInfo.startCursor)
    if (type) searchParams.set('type', type.toString())
    previousPageUrl = `/webhooks?${searchParams}`
}
```